src/containerapp/azext_containerapp/_utils.py

```diff
@@ -385,6 +392,158 @@ def parse_secret_flags(secret_list):
+    return bool(re.match(pattern, binding_name))
+
+
+def check_unique_bindings(cmd, service_connectors_def_list, service_bindings_def_list, resource_group_name, name):
+    linker_client = get_linker_client(cmd)
+    containerapp_def = None
+
+    try:
+        containerapp_def = ContainerAppClient.show(cmd=cmd, resource_group_name=resource_group_name, name=name)
+    except:  # pylint: disable=bare-except
+        pass
+    all_bindings = []
+
+    if containerapp_def:
+        managed_bindings = linker_client.linker.list(resource_uri=containerapp_def["id"])
+        service_binds = containerapp_def["properties"].get("template", {}).get("serviceBinds", [])
+
+        if managed_bindings:
+            all_bindings.append(item.name for item in managed_bindings)
+        if service_binds:
+            all_bindings.append(item["name"] for item in service_binds)
+
+    service_binding_names = [service_bind["name"] for service_bind in service_bindings_def_list]
+    linker_names = [connector["linker_name"] for connector in service_connectors_def_list]
+
+    all_bindings_set = set(all_bindings)
+    service_binding_names_set = set(service_binding_names)
+    linker_names_set = set(linker_names)
+
+    if len(all_bindings_set | service_binding_names_set | linker_names_set) != len(all_bindings_set) + len(
+            service_binding_names_set) + len(linker_names_set):
+        # There are duplicate elements across the lists
+        return False
+    elif len(all_bindings_set) + len(service_binding_names_set) + len(linker_names_set) != len(all_bindings) + len(
+            service_binding_names) + len(linker_names):
+        # There are duplicate elements within one or more of the lists
+        return False
+    else:
+        # There are no duplicate elements among the lists or within any of the lists
+        return True
+
+
+def parse_service_bindings(cmd, service_bindings_list, resource_group_name, name):
+    # Make it return both managed and dev bindings
+    service_bindings_def_list = []
+    service_connector_def_list = []
+
+    for service_binding_str in service_bindings_list:
+        parts = service_binding_str.split(",")
+        arg_dict = {}
+
+        for part in parts:
+            key_value = part.split("=")
+
+            if len(key_value) == 1:
+                # This means we don't have comma separated args
+                pass
+            else:
+                arg_dict[key_value[0]] = key_value[1]
+
+        service_binding = parts[0].split(':')
+        service_name = service_binding[0]
+
+        if len(service_binding) == 1:
+            binding_name = service_name
+        else:
+            binding_name = service_binding[1]
+
+        if not validate_binding_name(binding_name):
+            raise ValidationError("The Binding Name can only contain letters, numbers (0-9), periods ('.'), "
+                                  "and underscores ('_'). The length must not be more than 60 characters.")
```

Comment:
Could you please use the more specific error type `InvalidArgumentValueError`?
```suggestion
            raise InvalidArgumentValueError("The Binding Name can only contain letters, numbers (0-9), periods ('.'), "
                                  "and underscores ('_'). The length must not be more than 60 characters.")
```

Reply:
Fixed